modules/core/src/main/java/org/locationtech/jts/noding/ScaledNoder.java

```diff
@@ -71,15 +73,12 @@ public void computeNodes(Collection inputSegStrings)
 
   private Collection scale(Collection segStrings)
   {
-//  	System.out.println("Scaled: scaleFactor = " + scaleFactor);
-    return CollectionUtil.transform(segStrings,
-                                    new CollectionUtil.Function() {
-      public Object execute(Object obj) {
-        SegmentString ss = (SegmentString) obj;
-        return new NodedSegmentString(scale(ss.getCoordinates()), ss.getData());
-      }
-                                    }
-      );
+    List nodedSegmentStrings = new ArrayList();
+    for (Iterator i = segStrings.iterator(); i.hasNext(); ) {
```

Comment:
Can the transpiler handle Java 5 for/in syntax?  I think that will avoid the cast, and is a little simpler.

Reply:
Not currently (AFAIK there is no for/in use currently in JTS) but I would think it should not be that hard to implement. 

But, since `segStrings` is a generic `Collection` I don't see how I can avoid casting anyway?